ddl/ddl_api.go

```diff
@@ -1731,9 +1731,9 @@ func buildTableInfoWithStmt(ctx sessionctx.Context, s *ast.CreateTableStmt, dbCh
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	setTemporaryType(tbInfo, s.TemporaryKeyword)
 	switch s.TemporaryKeyword {
```

Comment:
This `switch case` is also needed for `create temporary tables like`.
E.g. If `EnableGlobalTemporaryTable` is disabled, or `OnCommitDelete` is omitted, an error is reported.
So you need to put this into `setTemporaryType`.